types/accessRequestSearchResponse.js

```diff
@@ -0,0 +1,61 @@
+const AccessRequest = require('./accessRequest')
+const Serializable = require('./serializable')
+
+class AccessRequestSearchResponse extends Serializable {
+  constructor(accessRequests, nextToken) {
+    super()
+    this.accessRequests = accessRequests
+    this.nextToken = nextToken
+  }
+
+  serializable() {
+    let serializedAccessRequests = []
+    for (const accessRequest of this.accessRequests) {
+      serializedAccessRequests.push(accessRequest.serializable())
+    }
+    let toSerialize = {
+      access_requests: serializedAccessRequests,
+      next_token: this.nextToken,
+    }
+    const serializedKeys = Object.keys(toSerialize)
+    for (const key of serializedKeys) {
+      if (toSerialize[key] === null) {
+        delete toSerialize[key]
+      }
+    }
+    return toSerialize
+  }
+  static async decode(json) {
+    const rawAccessRequests =
+      json.access_requests === undefined ? [] : json.access_requests
+    let accessRequests = []
+    for (const rawAccessRequest of rawAccessRequests) {
+      let accessRequest = new AccessRequest()
```

Comment:
I tried the much simpler

`const decodedAccessRequest = await AccessRequest.decode(rawAccessRequest)`

 but could not avoid getting told that "AccessRequest" is undefined, even though its literally quite usable in this line... 🤷🏽‍♂️ please tell me how to fix

Reply:
🤔

Reply:
my code doesn't seem to complain about this. was it just the test that failed in that way?

can you try to replace the whole for loop with
```js
const accessRequests = rawAccessRequests.map((ar) => AccessRequest.decode(ar))
```

Reply:
Results in
```
crypto (ignored) 15 bytes [optional] [built] [code generated]
webpack 5.36.2 compiled successfully in 777 ms

> @toznysecure/sdk@1.3.7 test
> jest "identity" "-t" "access"

 FAIL  __tests__/identity.test.js
  Tozny identity client
    ✓ it can create an access request (68 ms)
    ✕ it can search for all self created access requests (104 ms)
    ✓ it can describe a created access request (72 ms)
    ✓ it can delete a created access request (66 ms)
    ○ skipped can use refresh tokens to extend its session
    ○ skipped can do identity look ups based on emails
    ○ skipped can do identity look ups based on usernames
    ○ skipped can do identity look ups on fake username
    ○ skipped can do an identity look up based on email
    ○ skipped can do an identity look up based on username
    ○ skipped it can get private realm info

  ● Tozny identity client › it can search for all self created access requests

    expect(received).toEqual(expected) // deep equality

    Expected: ArrayContaining [185, 186]
    Received: [undefined, undefined, undefined]

      178 |       (accessRequest) => accessRequest.id
      179 |     )
    > 180 |     expect(searchResultAccessRequestIDs).toEqual(
          |                                          ^
      181 |       expect.arrayContaining([firstAccessRequest.id, secondAccessRequest.id])
      182 |     )
      183 |   })

      at Object.<anonymous> (__tests__/identity.test.js:180:42)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 7 skipped, 3 passed, 11 total
Snapshots:   0 total
Time:        2.297 s
Ran all test suites matching /identity/i with tests matching "access".
[octo@ValleyOfTheForge js-sdk]$ 
```

Reply:
```
 console.log
    AccessRequestSearchResponse {
      accessRequests: [
        Promise { [AccessRequest] },
        Promise { [AccessRequest] },
        Promise { [AccessRequest] }
      ],
      nextToken: 0
    }

      at Object.<anonymous> (__tests__/identity.test.js:177:13)
```

definetely looks like a noob await error I ran into and expect us to be able resolve in a few minutes of pairing